fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/api/tool/attribute/v1/DynamicAttributeTool.java

```diff
@@ -48,7 +69,28 @@
 	 * @return The mining speed multiplier of the item. 8.0 is equal to a diamond pick.
 	 */
 	//TODO: nullable on user once we have an official @Nullable annotation in
-	float getMiningSpeedMultiplier(ItemStack stack, LivingEntity user);
+	default float getMiningSpeedMultiplier(ItemStack stack, LivingEntity user) {
+		return this instanceof ToolItem ? ((ToolItem) this).getMaterial().getMiningSpeed() : 1f;
+	}
+
+	/**
+	 * Determines the mining speed multiplier of the passed stack, which is one factor in overall mining speed.
+	 *
+	 * @param state The block to mine.
+	 * @param stack The stack to check on.
+	 * @param user The current user of the tool, or null if there isn't any.
+	 * @return The mining speed multiplier of the item. 8.0 is equal to a diamond pick.
+	 */
+	//TODO: nullable on user once we have an official @Nullable annotation in
+	default float getMiningSpeedMultiplier(BlockState state, ItemStack stack, LivingEntity user) {
+		if (this instanceof ShearsItem) {
+			Block block = state.getBlock();
+			if (block == Blocks.COBWEB || state.matches(BlockTags.LEAVES)) return 15f;
+			if (state.matches(BlockTags.WOOL)) return 5f;
+		}
```

Comment:
Why not put this code in the ShearsItem mixin?

Reply:
Done.